src/UsgsAstroFrameSensorModel.cpp

```diff
@@ -793,120 +796,150 @@ void UsgsAstroFrameSensorModel::replaceModelState(const std::string& stringState
+      }
 
-    // get detector_center
-    state["m_ccdCenter"][0] = getDetectorCenterLine(isd, parsingWarnings);
-    state["m_ccdCenter"][1] = getDetectorCenterSample(isd, parsingWarnings);
+      // get detector_center
+      {
+        json jayson = isd.at("detector_center");
+        json sample = jayson.at("sample");
+        json line = jayson.at("line");
 
-    std::cerr << "Detector Center Parsed!" << std::endl;
+        state["m_ccdCenter"][0] = line;
+        state["m_ccdCenter"][1] = sample;
 
-    // get radii
-    state["m_minorAxis"] = getSemiMinorRadius(isd, parsingWarnings);
-    state["m_majorAxis"] = getSemiMajorRadius(isd, parsingWarnings);
+        std::cerr << "Detector Center Parsed!" << std::endl;
+      }
 
-    std::cerr << "Target Radii Parsed!" << std::endl;
+      // get radii
+      {
+        json jayson = isd.at("radii");
+        json semiminor = jayson.at("semiminor");
+        json semimajor = jayson.at("semimajor");
+        json unit = jayson.at("unit");
 
-    // get reference_height
-    state["m_minElevation"] = getMinHeight(isd, parsingWarnings);
-    state["m_maxElevation"] = getMaxHeight(isd, parsingWarnings);
+        unit = unit.get<std::string>();
+        state["m_minorAxis"] = metric_conversion(semiminor.get<double>(), unit);
+        state["m_majorAxis"] = metric_conversion(semimajor.get<double>(), unit);
 
-    std::cerr << "Reference Height Parsed!" << std::endl;
+        std::cerr << "Target Radii Parsed!" << std::endl;
+      }
 
-    state["m_ephemerisTime"] = getCenterTime(isd, parsingWarnings);
-    state["m_nLines"] = getTotalLines(isd, parsingWarnings);
-    state["m_nSamples"] = getTotalSamples(isd, parsingWarnings);
+      // get reference_height
+      {
+        json reference_height = isd.at("reference_height");
+        json maxheight = reference_height.at("maxheight");
+        json minheight = reference_height.at("minheight");
+        json unit = reference_height.at("unit");
 
-    state["m_iTransL"] = getFocal2PixelLines(isd, parsingWarnings);
-    state["m_iTransS"] = getFocal2PixelSamples(isd, parsingWarnings);
+        unit = unit.get<std::string>();
+        state["m_minElevation"] = metric_conversion(minheight.get<double>(), unit);
+        state["m_maxElevation"] = metric_conversion(maxheight.get<double>(), unit);
 
-    // We don't pass the pixel to focal plane transformation so invert the
-    // focal plane to pixel transformation
-    try {
+        std::cerr << "Reference Height Parsed!" << std::endl;
+      }
+
+      state["m_ephemerisTime"] = isd.at("center_ephemeris_time");
+      state["m_nLines"] = isd.at("image_lines");
+      state["m_nSamples"] = isd.at("image_samples");
+
+      state["m_iTransL"] = isd.at("focal2pixel_lines");
+
+      state["m_iTransS"] = isd.at("focal2pixel_samples");
+
+      // We don't pass the pixel to focal plane transformation so invert the
+      // focal plane to pixel transformation
```

Comment:
All of this code was un-done with the rebase. This whole chunk except for the distortion specific stuff needs to be reverted.